include/seqan3/search/dream_index/interleaved_bloom_filter.hpp

```diff
@@ -0,0 +1,501 @@
+    //!\endcond
+
+    //!\brief The underlying datatype to use.
+    using data_type = std::conditional_t<data_layout_mode_ == data_layout::uncompressed,
+                                         sdsl::bit_vector,
+                                         sdsl::sd_vector<>>;
+
+    //!\brief The number of bins specified by the user.
+    size_t bins{};
+    //!\brief The number of bins stored in the IBF (next multiple of 64 of `bins`).
+    size_t technical_bins{};
+    //!\brief The size of each bin in bits.
+    size_t bin_size_{};
+    //!\brief The number of 64-bit integers needed to store `bins` many bits (e.g. `bins = 50` -> `bin_words = 1`).
+    size_t bin_words{};
+    //!\brief The number of hash functions.
+    size_t hash_funs{};
+    //!\brief The bitvector.
+    data_type data{};
+    //!\brief Precalculated seeds for hashing.
+    std::vector<size_t> hash_seeds{};
+    //!\brief The result buffer for a `bulk_contains()` query.
+    mutable binning_bitvector result_buffer{};
+
+    /*!\brief Perturbs a value and fits it into the vector.
+     * \param h The value to process.
+     * \sa https://probablydance.com/2018/06/16/
+     * \sa https://lemire.me/blog/2016/06/27
+     */
+    constexpr void hash_and_fit(size_t & h) const
+    {
+        h *= 11400714819323198485ULL; // = 2^64 / golden_ration, to expand h to 64 bit range
+        // Use fastrange (integer modulo without division) if possible.
+#ifdef __SIZEOF_INT128__
+        h = static_cast<uint64_t>((static_cast<__uint128_t>(h) * static_cast<__uint128_t>(bin_size_)) >> 64);
+#else
+        h %= bin_size_;
+#endif
+        h *= technical_bins;
+    }
+
+public:
+    //!\brief Indicates whether the Interleaved Bloom Filter is compressed.
+    static constexpr data_layout data_layout_mode = data_layout_mode_;
+
+    /*!\name Constructors, destructor and assignment
+     * \{
+     */
+    interleaved_bloom_filter() = default; //!< Defaulted.
+    interleaved_bloom_filter(interleaved_bloom_filter const &) = default; //!< Defaulted.
+    interleaved_bloom_filter & operator=(interleaved_bloom_filter const &) = default; //!< Defaulted.
+    interleaved_bloom_filter(interleaved_bloom_filter &&) = default; //!< Defaulted.
+    interleaved_bloom_filter & operator=(interleaved_bloom_filter &&) = default; //!< Defaulted.
+    ~interleaved_bloom_filter() = default; //!< Defaulted.
+
+    /*!\brief Construct an uncompressed Interleaved Bloom Filter.
+     * \param bins_ The number of bins.
+     * \param bin_size__ The bitvector size.
+     * \param funs The number of hash functions. Default 2.
+     *
+     * \attention This constructor can only be used to construct **uncompressed** Interleaved Bloom Filters.
+     *
+     * \details
+     *
+     * ### Example
+     *
+     * \include test/snippet/search/dream_index/interleaved_bloom_filter_constructor.cpp
+     */
+    interleaved_bloom_filter(seqan3::bin_count bins_,
+                             seqan3::bin_size bin_size__,
```

Comment:
two underscores :see_no_evil: Maybe just call it size here in this very small context since the strong type is already expressive enough?
```suggestion
                             seqan3::bin_size size,
```